### ISSUE TITLE
mount.mounted does not add entry in fstab if the mount can't be done.

### ISSUE THREAD

**Issue by markhor75** [saltstack/salt#36837]
### Description of Issue/Question

mount.mounted does not add entry in fstab if the mount can't be done. 
workaround : set mount option to False

### Setup

This sls doesn't work :

> /media/mydir:
>   mount.mounted:
>     - device: nas.acme.tld:/volume/mydir
>     - fstype: nfs
>     - mkmnt: True
>     - opts:
>       - defaults

With this sls it works :

> /media/mydir:
>   mount.mounted:
>     - device: nas.acme.tld:/volume/mydir
>     - fstype: nfs
>     - mkmnt: True
>     - mount: False
>     - persist: True
>     - opts:
>       - defaults

### Versions Report
**Server :** 
> Salt Version:
>            Salt: 2016.3.3
>  
> Dependency Versions:
>            cffi: 0.8.6
>        cherrypy: Not Installed
>        dateutil: 2.2
>           gitdb: 0.5.4
>       gitpython: 0.3.2 RC1
>           ioflo: Not Installed
>          Jinja2: 2.7.3
>         libgit2: Not Installed
>         libnacl: Not Installed
>        M2Crypto: Not Installed
>            Mako: Not Installed
>    msgpack-pure: Not Installed
>  msgpack-python: 0.4.2
>    mysql-python: 1.2.3
>       pycparser: 2.10
>        pycrypto: 2.6.1
>          pygit2: Not Installed
>          Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
>    python-gnupg: Not Installed
>          PyYAML: 3.11
>           PyZMQ: 14.4.0
>            RAET: Not Installed
>           smmap: 0.8.2
>         timelib: Not Installed
>         Tornado: 4.2.1
>             ZMQ: 4.0.5
>  
> System Versions:
>            dist: debian 8.6 
>         machine: x86_64
>         release: 3.16.0-4-amd64
>          system: Linux
>         version: debian 8.6 
> 

**Client :**
> Ubuntu 14.04.4 LTS 
> salt-minion 2015.5.3 (Lithium)

**Comment by Ch3LL** [comment]
@markhor75 i believe your workaround exists because `mount: False` won't even attempt to mount it will simply add to fstab as long as persist: True is also set, which I believe is what you actually want in this scenario. Maybe i'm missing something? Do you want the default behavior to add the mount information even if their is an error?

**Comment by xcorvis** [comment]
I'm running into a related issue and it would have been great to find this info in the documentation. I think there's an argument to be made that this is not intended behavior, but in the mean time just adding it to the mount docs would be helpful.

**Comment by Ch3LL** [comment]
can you clarify what was confusing and what we can document? The combination of mount: False and persist?

**Comment by xcorvis** [comment]
Right, just that. If the mount fails, the persistent command will not be written to the fstab, but you can still force it to write if mount is set to false.

**Comment by Ch3LL** [comment]
i'm starting to wonder if this is a bug or if we could just add an option to ensure it adds to fstab even if the mount fails. ping @saltstack/team-core opinions?

**Comment by DmitryKuzmenko** [comment]
`persist` is `True` by default, so the only difference in these states is `mount` option value. But as `mount.mounted` documentation says the `persist` option declares do we add the option to fstab or not.
But if `mount` is `True` and the mountpoint couldn't be mounted the state execution fails immediately and mountpoint will not be added to fstab. As I can see by the code here we have this case.
@xcorvis could you please confirm this?

**Comment by xcorvis** [comment]
I'm sorry, I'm having trouble understanding. What do you want me to confirm?
